controllers/k8ssandra/secrets.go

```diff
@@ -41,6 +41,10 @@ func SuperuserSecretName(kc *api.K8ssandraCluster) string {
 }
 
 func (r *K8ssandraClusterReconciler) reconcileReaperSecrets(ctx context.Context, kc *api.K8ssandraCluster, logger logr.Logger) result.ReconcileResult {
+	if kc.Spec.Reaper != nil && kc.Spec.Reaper.ReaperRef.Name != "" {
+		logger.Info("ReaperRef points to an existing Reaper, we don't need a new Reaper CQL secret")
```

Comment:
I don't think it's as straightforward here. But perhaps I still don't understand things well enough.

We are here in the k8ssandra cluster reconciliation loop, and we're checking what the kc.reaper.Spec tells us.
We are deciding what to do with the CQL secret and the UI secret. We're not dealing with the JMX (except if the CQL and JMX secrets are one and the same).

We only want to reconcile(~create) the secrets if we're using a k8ssandra-cluster-specific Reaper.

We can have a k8ssandra-cluster-specific Reaper with local storage. In that case we don't need a CQL secret, only the UI secret. This is checked on L68.

I think that's why we cannot check for just the storage on L44. I'd really need the `isControlPlane()` function somewhere, but I cannot have that for other reasons. The ReaperRef presence is still my favourite alternative to that.

To sum up, I think I'd need clarification of:
* how the JMX secret comes into this?
* do we actually want to allow _bundled_ Reapers with local storage?

Reply:
> do we actually want to allow bundled Reapers with local storage?

Yes, why not? It's an efficient storage mechanism which doesn't require using Cassandra itself.

> how the JMX secret comes into this?

The cases where we need Reaper to create users is for:

- authenticated access with JMX enabled
- using cassandra as storage for Reaper

One could have a centralized Reaper using JMX to communicate with the Cassandra clusters. It's not the default, but it's a possibility.
In this case, we still need a user and its secret to be created.

It is just a suggestion though and if that feels like some complications we want to deal with later, I'm good with that.

Reply:
>> do we actually want to allow bundled Reapers with local storage?

> Yes, why not? It's an efficient storage mechanism which doesn't require using Cassandra itself.

I agree with this, so that's sorted. 

About the JMX, I still don't get why we worry about it here when the `reconcileReaperSecrets` does not mention JMX secrets at all.

I'm indeed leaning towards not accepting this suggestion and keeping the condition as it is.

Reply:
ok then 👍